src/core/qgsmaphittest.cpp

```diff
@@ -109,6 +109,21 @@ void QgsMapHitTest::runHitTestLayer( QgsVectorLayer *vl, SymbolSet &usedSymbols,
   const bool moreSymbolsPerFeature = r->capabilities() & QgsFeatureRenderer::MoreSymbolsPerFeature;
   r->startRender( context, vl->fields() );
 
+  // shortcut early if we know that there's nothing visible
+  if ( r->canSkipRender() )
```

Comment:
Am I correctly assuming this is only safe to call after startRender?

Reply:
yes (and the docs note that explicitly)